ocaml/xapi/helpers.ml

```diff
@@ -1654,8 +1654,9 @@ let retry ~__context ~doc ?(policy = Policy.standard) f =
       ) ;
       f ()
     with
-    | Api_errors.Server_error (code, _ :: _) as e
+    | Api_errors.Server_error (code, _) as e
     when code = Api_errors.other_operation_in_progress
+         || code = Api_errors.get_updates_in_progress
```

Comment:
Rather than adding a special case here, could we move the following in xapi_pool_helpers.ml
```
  ; (`get_updates, Api_errors.get_updates_in_progress)
```
from the `blocking_ops` list to `wait_ops`? Then we can also delete the definition of `Api_errors.get_updates_in_progress` completely.

Reply:
Thanks. I oversaw the `wait_ops` and `Helpers.Early_wakeup.broadcast` in `with_pool_operation`.
On the other hand, indeed, it would be odd if the `Api_errors.get_updates_in_progress` would not be reported to users but still an error.